File: Backend/TranscomaAPI/Comun/Entidades/Direccion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    /// <summary>
    /// Entidad para almacenar las direcciones
    /// de envio y recepcion de carga
    /// </summary>
    public class Direccion: Entidad
    {
        private string _codigoZip;
        private string _ciudad;
        private string _provincia;
        private string _estado;
        private Pais _pais;

        public Direccion(int id)
        {
            Id = id;
        }

        public Direccion(string codigoZip, string ciudad, string provincia, string estado, Pais pais)
        {
            _codigoZip = codigoZip;
            _ciudad = ciudad;
            _provincia = provincia;
            _estado = estado;
            _pais = pais;
        }

        public string CodigoZip { get => _codigoZip; set => _codigoZip = value; }
        public string Ciudad { get => _ciudad; set => _ciudad = value; }
        public string Provincia { get => _provincia; set => _provincia = value; }
        public string Estado { get => _estado; set => _estado = value; }
        public Pais Pais { get => _pais; set => _pais = value; }
    }
}

File: frontend/src/app/models/Direccion.ts
import { EntidadBase } from "./EntidadBase";

export class Direccion extends EntidadBase {

    private _codigoZip: string ;
    private _ciudad: string ;
    private _provincia: string ;
    private _estado: string ;
    private _pais: string;

    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$CodigoZip = data.codigoZip;
            this.$Ciudad = data.ciudad;
            this.$Provincia = data.provincia;
            this.$Estado = data.estado;
            this.$Pais = data.pais;
        }
    }

    /** Getters y setters */
    public get $CodigoZip(): string {
        return this._codigoZip;
    }

    public set $CodigoZip(value: string) {
        this._codigoZip = value;
    }

    public get $Ciudad(): string{
        return this._ciudad;
    }

    public set $Ciudad(value: string) {
        this._ciudad = value;
    }

    public get $Provincia(): string {
        return this._provincia;
    }

    public set $Provincia(value: string) {
        this._provincia = value;
    }

    public get $Estado(): string {
        return this._estado;
    }

    public set $Estado(value: string) {
        this._estado = value;
    }

    public get $Pais(): string {
        return this._pais;
    }

    public set $Pais(value: string) {
        this._pais = value;
    }
}
